### PR TITLE
Return clients by reference

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Fixed up a headergroup test

### CODE REVIEW

File: test/autorest/arraygroup/arraygroup_test.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 func newArrayClient() ArrayClient {
-	return NewArrayClient(NewDefaultConnection(nil))
+	return *NewArrayClient(NewDefaultConnection(nil))
```
Comment (catalinaperalta) [inline test/autorest/arraygroup/arraygroup_test.go:18]: Is this on purpose?

Comment (jhendrixMSFT) [reply test/autorest/arraygroup/arraygroup_test.go:18]: Good catch, copy/pasta error.

Comment (jhendrixMSFT) [reply test/autorest/arraygroup/arraygroup_test.go]: Fixed.

**Review by catalinaperalta — APPROVED** [review]
LGTM

**Pull request merged**